virl2_client/models/auth_management.py

```diff
@@ -0,0 +1,391 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import annotations
+
+import time
+from enum import Enum
+from typing import Any, Optional
+
+from httpx import Client
+
+from .resource_pools import ResourcePool
+
+
+class AuthMethod(Enum):
+    LOCAL = "local"
+    LDAP = "ldap"
+
+
+_BASE_URL = "system/auth"
+
+
+class AuthManagement:
+    _METHOD_URL = _BASE_URL + "/config"
+
+    def __init__(self, session: Client, auto_sync=True, auto_sync_interval=1.0):
+        self.auto_sync = auto_sync
+        self.auto_sync_interval = auto_sync_interval
+        self._last_sync_time = 0.0
+        self._session = session
+        self._method = AuthMethod.LOCAL
+        self._managers: dict[AuthMethod, Optional[AuthMethodManager]] = {
+            AuthMethod("local"): None,
+            AuthMethod("ldap"): LDAPManager(self),
+        }
+
+    def sync_if_outdated(self) -> None:
+        timestamp = time.time()
+        if (
+            self.auto_sync
+            and timestamp - self._last_sync_time > self.auto_sync_interval
+        ):
+            self.sync()
+
+    def sync(self) -> None:
+        settings: dict = self._session.get(self._METHOD_URL).json()
+        self._method = AuthMethod(settings.pop("method"))
+        self._last_sync_time = time.time()
+        # Since right now we also get the other settings along with the method, we
+        # might as well save a request and store said settings in the relevant manager
+        if (manager := self._managers[self._method]) is not None:
+            manager._settings = settings
+            manager._last_sync_time = time.time()
+
+    @property
+    def method(self) -> AuthMethod:
+        self.sync_if_outdated()
+        return self._method
+
+    @method.setter
+    def method(self, value: AuthMethod | str):
+        value = AuthMethod(value)
+        self._session.put(self._METHOD_URL, json={"method": value.value})
+        self._method = value
+
+    @property
+    def manager(self) -> AuthMethodManager:
+        """
+        The manager for the current auth method.
+        """
+        # TODO: should we allow users to also configure currently inactive auth methods?
+        #  not sure how useful that might or might not be, but if they can then that
+        #  changes the answer to the question in AuthMethodManager.update_setting
+        self.sync_if_outdated()
+        return self._managers[self._method]
+
+
+class AuthMethodManager:
+    _METHOD = ""
```

Comment:
shouldn't this be None or a value for the AuthMethod enum?

Reply:
A string is more convenient (which matters more since this is private) and honestly it does not matter what value we set as default, it's going to be overwritten by subclass anyway